lms/djangoapps/grades/api/views.py

```diff
@@ -226,3 +239,161 @@ def get(self, request, course_id, **kwargs):
         if isinstance(course, Response):
             return course
         return Response(GradingPolicySerializer(course.raw_grader, many=True).data)
+
+
+@view_auth_classes()
+class GradeReportBySection(GenericAPIView):
```

Comment:
class `GradeReportBySection`, `GradeReporByTaskId` and `GradeReportEnhancedProblemGrade` get function is the same code. The only thing that changes is the `submit_report_type`. Can we have a generic function instead?